--- local_scan.h
+++ local_scan.h
@@ -196,7 +196,19 @@
 
 extern int     child_close(pid_t, int);
 extern void    debug_printf(const char *, ...) PRINTF_FUNCTION(1,2);
-extern uschar *expand_string(uschar *);
+#define DKIM2_LOCAL_SCAN_EXPAND_API 1
+extern const uschar *expand_string_2(const uschar *, BOOL *);
+
+static inline uschar *
+expand_nc_string(uschar *value)
+{ return US expand_string_2(value, NULL); }
+
+static inline const uschar *
+expand_c_string(const uschar *value)
+{ return expand_string_2(value, NULL); }
+
+#define expand_string(value) _Generic((value), uschar *: expand_nc_string, \
+  const uschar *: expand_c_string)(value)
 extern void    header_add(int, const char *, ...);
 extern void    header_add_at_position(BOOL, uschar *, BOOL, int, const char *, ...);
 extern void    header_remove(int, const uschar *);
--- functions.h
+++ functions.h
@@ -240,16 +240,18 @@
 extern uschar *expand_file_big_buffer(const uschar *);
 
+#ifndef DKIM2_LOCAL_SCAN_EXPAND_API
 extern const uschar *expand_string_2(const uschar *, BOOL *);
 
 static inline uschar * expand_nc_string(uschar * s)
 { return US expand_string_2(s, NULL); }
 static inline const uschar * expand_c_string(const uschar * s)
 { return expand_string_2(s, NULL); }
 
 /* A macro that picks which function to use depending on the type of the arg */
 #define expand_string(X) _Generic((X),     \
 	      uschar *:		expand_nc_string, \
 	      const uschar *:	expand_c_string   \
 	      )(X)
+#endif
 
 extern BOOL   expand_string_nonempty(const uschar *);
