--- local_scan.h
+++ local_scan.h
@@ -2,7 +2,7 @@
 *     Exim - an Internet mail transport agent    *
 *************************************************/
 
-/* Copyright (c) The Exim Maintainers 2020 - 2025 */
+/* Copyright (c) The Exim Maintainers 2020 - 2024 */
 /* Copyright (c) University of Cambridge 1995 - 2020 */
 /* See the file NOTICE for conditions of use and distribution. */
 /* SPDX-License-Identifier: GPL-2.0-or-later */
@@ -22,14 +22,13 @@
 store.c
 */
 
-#ifndef LOCAL_SCAN_H
-#define LOCAL_SCAN_H
 
 /* Some basic types that make some things easier, the Exim configuration
 settings, and the store functions. */
 
 #include <stdarg.h>
 #include <sys/types.h>
+#pragma GCC visibility push(default)
 #include "config.h"
 #include "mytypes.h"
 #include "store.h"
@@ -179,6 +178,9 @@
 extern BOOL    host_checking;          /* Set when checking a host */
 extern uschar *interface_address;      /* Interface for incoming call */
 extern int     interface_port;         /* Port number for incoming call */
+#ifdef DLOPEN_LOCAL_SCAN
+extern uschar *local_scan_path;
+#endif
 extern uschar *message_id;             /* Internal id of message being handled */
 extern uschar *received_protocol;      /* Name of incoming protocol */
 extern int     recipients_count;       /* Number of recipients */
@@ -213,7 +215,7 @@
 extern BOOL    receive_remove_recipient(const uschar *);
 extern uschar *rfc2047_decode(uschar *, BOOL, const uschar *, int, int *,
 			      uschar **);
-extern int     smtp_fflush(BOOL);
+extern int     smtp_fflush(void);
 extern void    smtp_printf(const char *, BOOL, ...) PRINTF_FUNCTION(1,3);
 extern void    smtp_vprintf(const char *, BOOL, va_list);
 
@@ -249,5 +251,6 @@
 extern pid_t    child_open_function(uschar **, uschar **, int, int *, int *, BOOL, const uschar *);
 #endif
 
-#endif	/* whole file */
+#pragma GCC visibility pop
+
 /* End of local_scan.h */
